sparkle/src/components/DataTable.tsx

```diff
@@ -101,24 +99,34 @@ export function DataTable<TData extends TBaseData>({
         }
       : undefined;
 
+  const onSortingChange =
+    sorting && setSorting
+      ? (updater: Updater<SortingState>) => {
+          const newValue =
+            typeof updater === "function" ? updater(sorting) : updater;
+          setSorting(newValue);
+        }
+      : undefined;
+
   const table = useReactTable({
     data,
     columns,
     rowCount: totalRowCount,
     manualPagination: isServerSidePagination,
-    onSortingChange: setSorting,
+    onSortingChange: onSortingChange,
     getCoreRowModel: getCoreRowModel(),
-    getSortedRowModel: getSortedRowModel(),
+    getSortedRowModel: sorting ? getSortedRowModel() : undefined,
```

Comment:
We still want to support client side sorting oob. the getSortedModelRow is only used if using client side sort (and it's mandatory), so I would invert the test here. 
To enable client-side sorting it you must not pass `sorting` in the state nor `onSortingChange` (even as undefined). Also client-side should be automatically disabled if pagination is enabled.